types/setting.go

```diff
@@ -757,6 +757,9 @@ func ValidateInitSetting(name, value string) (err error) {
 	if definition.Required && value == "" {
 		return fmt.Errorf("required setting %v shouldn't be empty", sName)
 	}
+	if definition.ReadOnly && value != "" {
+		return fmt.Errorf("setting %v is read-only", sName)
+	}
```

Comment:
For the setting CR creation, this check should not be applied. In other words, this check is not for **init** setting validation

Reply:
I think the function name should be reworded, remove the `init` term so it won't be confused anymore.
After that, the function `ValidateSetting` would be called by
- CreateOrUpdateSetting: the RESTful API endpoint calls create/update the setting.
- validateSetting: the validating admission webhook calls create/update the Setting CR.

Reply:
Oh, OK. The read-only CR should neither be created/customized by users during installation nor be edited for an existing Longhorn. Putting the check here make sense, and you can rename the function if necessary

Reply:
Okay, I choose to rename the function.

Reply:
Sorry, I just realized adding the checking for the ReadOnly fields is harder than I was thought.

Right now, I can't find a way to distinguish that whether the create/update comes from longhorn-manager _or_ the user.

For example, the longhorn-manager will create/update the ReadOnly field `default-engine-image`. Every time when the default-engine-image changes, the longhorn-manager updates the value
- https://github.com/longhorn/longhorn-manager/blob/5b64adfaa300d2765cfeba232b883cf35b28e1bc/app/daemon.go#L221-L228
- https://github.com/longhorn/longhorn-manager/blob/5b64adfaa300d2765cfeba232b883cf35b28e1bc/manager/setting.go#L42-L47
- https://github.com/longhorn/longhorn-manager/blob/5b64adfaa300d2765cfeba232b883cf35b28e1bc/datastore/longhorn.go#L81-L82

Then, the longhorn-webhook server receives the update request and will verify it's a ReadOnly field and should be updated.
- https://github.com/longhorn/longhorn-manager/blob/87b7263df396dbdf9db9df00c51bc552767b816d/webhook/resources/setting/validator.go#L40-L47

Therefore, both the longhorn-manager _or_ the user creates/updates the ReadOnly field would thru. the same longhorn-webhook server validator.

Reply:
All right. Then let's remove the ReadOnly field check now...

Reply:
> All right. Then let's remove the ReadOnly field check now...

Removed.